warp-core/src/main/scala/com/workday/warp/TestIdImplicits.scala

```diff
@@ -0,0 +1,41 @@
+package com.workday.warp
+
+import java.lang.reflect.Method
+
+import com.workday.warp.common.utils.Implicits.DecoratedOptional
+import org.junit.jupiter.api.TestInfo
+import org.junit.jupiter.api.extension.ExtensionContext
+
+import scala.util.Try
+
+/**
+  * Ad-hoc polymorphism for constructing [[TestId]].
+  *
+  * Created by tomas.mccandless on 6/18/20.
+  */
+object TestIdImplicits {
+
+
+  /**
+    * Constructs a [[TestId]] from a [[TestInfo]].
+    *
+    * @param info
```

Comment:
Did you want to fill in the docs for testInfoIsTestId and extensionContextIsTestId?